lib/pipeline.js

```diff
@@ -1,3 +1,4 @@
+/* eslint no-param-reassign: ["error", { "props": false }] */
```

Comment:
I'd LOVE to be able to do this, but it's in our eslint rules.  If we want to do this, let's remove that rule globally, not per file.